tests/test_packages/test_skills/test_generic_buyer/test_behaviours.py

```diff
@@ -0,0 +1,163 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains test case classes based on pytest for AEA end-to-end testing."""
+
+from pathlib import Path
+from typing import cast
+
+from aea.test_tools.test_skill import BaseSkillTestCase
+
+from packages.fetchai.connections.ledger.base import CONNECTION_ID as LEDGER_PUBLIC_ID
+from packages.fetchai.protocols.ledger_api.message import LedgerApiMessage
+from packages.fetchai.skills.generic_buyer.behaviours import GenericSearchBehaviour
+from packages.fetchai.skills.generic_buyer.strategy import GenericStrategy
+
+from tests.conftest import ROOT_DIR
+
+
+FETCHAI = "fetchai"
+
+
+class TestSkillBehaviour(BaseSkillTestCase):
+    """Test behaviours of generic buyer."""
+
+    path_to_skill = Path(ROOT_DIR, "packages", "fetchai", "skills", "generic_buyer")
+
+    def test_search_behaviour_setup_a(self):
+        """Test the setup method of the search behaviour."""
+        outbox = self.skill.skill_context.outbox
+        search_behaviour = cast(
+            GenericSearchBehaviour, self.skill.skill_context.behaviours.search
+        )
+
+        assert outbox.empty()
```

Comment:
In some sense this check is unnecessary as the test tool guarantees this

Reply:
It depends on what we decide (see the discussion above)
- If we set up the skill at the beginning of each test, then yes, that's true.
- If not, then that line works as a "sanity check". But I'd say it's not best practice, since if tests would be well written there shouldn't be the need for it.

Reply:
See below; I am in favour of clean skill reset for each test method.